apps/meteor/app/federation-v2/server/application/RoomServiceSender.ts

```diff
@@ -148,18 +148,50 @@ export class FederationRoomServiceSender {
 		}
 	}
 
-	public async canAddUsersToTheRoom(internalUser: IUser | string, internalRoom: IRoom): Promise<void> {
-		const newUserBeingAdded = typeof internalUser === 'string';
-		if (newUserBeingAdded) {
+	public async canAddUsersToTheRoom(internalUser: IUser | string, internalInviter: IUser, internalRoom: IRoom): Promise<void> {
+		if (!internalRoom.federated) {
 			return;
 		}
+		const tryingToAddNewFederatedUser = typeof internalUser === 'string';
+		if (tryingToAddNewFederatedUser) {
+			throw new Error('error-this-is-an-ee-feature');
+		}
 
-		if (!internalRoom.federated) {
+		const invitee = await this.rocketUserAdapter.getFederatedUserByInternalId((internalUser as IUser)._id);
+		const inviter = await this.rocketUserAdapter.getFederatedUserByInternalId((internalInviter as IUser)._id);
```

Comment:
ok, with the new validation if the param is not a string you can remove the type casting:
```suggestion
		const invitee = await this.rocketUserAdapter.getFederatedUserByInternalId(internalUser._id);
		const inviter = await this.rocketUserAdapter.getFederatedUserByInternalId(internalInviter._id);
```

Reply:
for sure

Reply:
> btw, I've found another similar type castings in this very same file (that are not part of this PR), if you could take a look on them as well, it would be good 😬

About the type castings, I'll address that in another PR.